lntest/itest/lnd_test.go

```diff
@@ -1363,129 +1360,170 @@ func testBasicChannelFunding(net *lntest.NetworkHarness, t *harnessTest) {
+					carolCommitType: carolCommitType,
+					daveCommitType:  daveCommitType,
+					fundReceiver:    true,
+				})
+				tests = append(tests, testCase{
+					carolCommitType:    carolCommitType,
+					daveCommitType:     daveCommitType,
+					rejectHtlcReceiver: true,
+				})
+			} else {
+				tests = append(tests, testCase{
+					carolCommitType: carolCommitType,
+					daveCommitType:  daveCommitType,
+				})
 			}
+		}
+	}
 
-			daveArgs := daveCommitType.Args()
-			dave, err := net.NewNode("Dave", daveArgs)
-			if err != nil {
-				t.Fatalf("unable to create new node: %v", err)
-			}
+test:
+	for _, test := range tests {
+		success := runBasicFundingFlow(
+			net, t, test.carolCommitType, test.daveCommitType,
+			test.fundReceiver, test.rejectHtlcReceiver,
+		)
+		if !success {
+			break test
+		}
+	}
+}
 
-			// Before we start the test, we'll ensure both sides
-			// are connected to the funding flow can properly be
-			// executed.
-			ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
-			err = net.EnsureConnected(ctxt, carol, dave)
-			if err != nil {
-				t.Fatalf("unable to connect peers: %v", err)
-			}
+func runBasicFundingFlow(
+	net *lntest.NetworkHarness,
+	t *harnessTest,
+	carolCommitType, daveCommitType commitType,
+	fundReceiver bool,
+	rejectHtlcReceiver bool) bool {
 
-			testName := fmt.Sprintf("carol_commit=%v,dave_commit=%v",
-				carolCommitType, daveCommitType)
+	t.t.Helper()
 
-			ht := t
-			carolCommitType := carolCommitType
-			daveCommitType := daveCommitType
-			success := t.t.Run(testName, func(t *testing.T) {
-				carolChannel, daveChannel, closeChan, err := basicChannelFundingTest(
-					ht, net, carol, dave, nil,
-				)
-				if err != nil {
-					t.Fatalf("failed funding flow: %v", err)
-				}
+	ctxb := context.Background()
 
-				// Both nodes should report the same commitment
-				// type.
-				chansCommitType := carolChannel.CommitmentType
-				if daveChannel.CommitmentType != chansCommitType {
-					t.Fatalf("commit types don't match, "+
-						"carol got %v, dave got %v",
-						carolChannel.CommitmentType,
-						daveChannel.CommitmentType,
-					)
-				}
+	// Based on the current tweak variable for Carol, we'll preferentially
+	// signal the legacy commitment format.  We do the same for Dave shortly
+	// below.
+	carolArgs := carolCommitType.Args()
+	carol, err := net.NewNode("Carol", carolArgs)
+	require.NoError(t.t, err)
 
-				// Now check that the commitment type reported
-				// by both nodes is what we expect. It will be
-				// the minimum of the two nodes' preference, in
-				// the order Legacy, Tweakless, Anchors.
-				expType := carolCommitType
+	// Each time, we'll send Carol a new set of coins in order to fund the
+	// channel.
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+	err = net.SendCoins(ctxt, btcutil.SatoshiPerBitcoin, carol)
+	require.NoError(t.t, err)
 
-				switch daveCommitType {
+	daveArgs := daveCommitType.Args()
+	if rejectHtlcReceiver {
```

Comment:
The test should also use this to determine if a test failure is acceptable. In the case where we don't fund the receiver, but then reject HTLCs for the receiver, and we're doing an anchor commitment funding, then we'd expect the funding flow to fail.